### PR TITLE
Fix daemon crash when groups list is empty

### PR DESCRIPTION
**Pull request by mariusSincovici**
When groups list is nil, return an error message to let the use know that the list is being fetched.

### CODE REVIEW

File: daemon/rpc_groups.go [hunk 1]
```diff
@@ -13,12 +13,30 @@ import (
 // Groups provides endpoint and autocompletion.
 func (r *RPC) Groups(ctx context.Context, in *pb.GroupsRequest) (*pb.Payload, error) {
 	var cfg config.Config
+	if r.cm == nil || r.dm == nil {
+		log.Println(internal.ErrorPrefix, "configuration Manager or DataManager is nil")
```
Comment (Savolro) [inline daemon/rpc_groups.go:17]: Typo: Choose one:
```
ConfigurationManager or DataManager
Configuration Manager or Data Manager
configuration manager or data manager
```

File: daemon/rpc_groups.go [hunk 2]
```diff
@@ -13,12 +13,30 @@ import (
...
 	if err := r.cm.Load(&cfg); err != nil {
 		log.Println(internal.ErrorPrefix, err)
+		return &pb.Payload{
+			Type: internal.CodeConfigError,
+		}, nil
+	}
+
+	groups := r.dm.GetAppData().GroupNames[in.GetObfuscate()][cfg.AutoConnectData.Protocol]
+	if groups == nil {
+		log.Println(internal.ErrorPrefix, "groups list are nil")
```
Comment (Savolro) [inline daemon/rpc_groups.go:32]: Typo. Change to one of:
```
group list is nil
group list is empty
```

**Review by Savolro — APPROVED** [review]

File: daemon/rpc_groups.go [hunk 3]
```diff
@@ -13,12 +13,30 @@ import (
 // Groups provides endpoint and autocompletion.
 func (r *RPC) Groups(ctx context.Context, in *pb.GroupsRequest) (*pb.Payload, error) {
 	var cfg config.Config
+	if r.cm == nil || r.dm == nil {
```
Comment (EmilijusS) [inline daemon/rpc_groups.go:16]: I don't think this check is necessary actually. At least we are not doing such checks elsewhere. Of course there's a theoretical possibility of a null pointer panic, but in practice this struct has a "constructor" so the actual possibility of them being nil is close to zero unless you are doing something unwise deliberately.
My opinion in general is that checking every time if every used interface is nil is just cumbersome, and in practice at least I never got burned by not checking them :smile:

Comment (mariusSincovici) [reply daemon/rpc_groups.go:16]: yes, this only happened while writing the tests. I'll remove the check in this case.

**Review by EmilijusS — APPROVED** [review]

**Pull request merged**